### PR TITLE
Support Storybook 7 

### LINKED ISSUE #3: Using Play inside test files
I noticed there isn't an option to run interactions from within a test file with the `Play` method. I have it running on a forked branch, but can't get it to actually execute userEvents. Is there any proposed update with this feature on the roadmap? I noticed it on the React version.

### LINKED ISSUE #8: Support story format CSF v3.0
I'm write the stories with new CSF v3.0 format, but `testing-vue3` support only the previous CSF format. When will available the support to the new format?

Component example:
```
export default {
  title: "Components/Button",
  component: Button
} as Meta;

export const Basic: Story = {
  parameters: {
    docs: {
      source: {
        code: `<Button>Example</Button>`
      },
      description: {
        component: "Customizable button with content."
      }
    }
  }
};
```

I get this error, when trying to use CSF 3.0 stories with `composeStories`.

```
 packages/components/src/button/Button.test.ts [ packages/components/src/button/Button.test.ts ]
Error: Cannot compose story due to invalid format. @storybook/testing-vue expected a function but received object instead.
 ❯ composeStory node_modules/.pnpm/@storybook+testing-vue3@0.0.2_i2pafdhxvhztb374fr7cj3wsfq/node_modules/@storybook/testing-vue3/dist/testing-vue3.cjs.development.js:140:11
 ❯ node_modules/.pnpm/@storybook+testing-vue3@0.0.2_i2pafdhxvhztb374fr7cj3wsfq/node_modules/@storybook/testing-vue3/dist/testing-vue3.cjs.development.js:194:23
 ❯ Proxy.composeStories node_modules/.pnpm/@storybook+testing-vue3@0.0.2_i2pafdhxvhztb374fr7cj3wsfq/node_modules/@storybook/testing-vue3/dist/testing-vue3.cjs.development.js:191:49
 ❯ packages/components/src/button/Button.test.ts:8:18
      6| import { Button } from "./index";
      7| 
      8| const { Basic } = composeStories(stories);
       |                  ^
      9| 
     10| describe("Basic tests", () => {
```

### Environment

 - OS: MacOS 
 - Node.js version: v16.13.2
 - NPM version: 8.1.2
 
### Additional context
{ ...
"@storybook/vue3": "^6.5.9",
 "@storybook/testing-vue3": "^0.0.2",
"@storybook/builder-vite": "^0.2.0",
"@testing-library/vue": "^6.6.1",
"vitest": "^0.18.1",
...
}

### LINKED ISSUE #10: Add support for storybook 7
No `next` version seems to have been created for this library, which breaks our whole test suite when attempting to upgrade to storybook 7

I've had a look at the issue a little bit, and after updating `addons` to use a named export (#9) there is a bunch of typing changes that I can't quite figure out.

This has been done for the react equivalent repo already here https://github.com/storybookjs/testing-react/pull/120 and a lot of it would be re-usable here as far as I can tell

### LINKED ISSUE #11: composeStories() ignores excludeStories
If you have a named Export in a CSF Story, tests will fail, even if it is excluded with `excludeStories: /.*Props$/`:

```
export const PrimaryProps: ButtonType = {
  size: 'medium',
  variant: 'primary',
  text: 'my happy primary button',
}
export const Primary = Template.bind({})
Primary.args = PrimaryProps
```

`Error: Cannot compose story due to invalid format. @storybook/testing-vue expected a function but received object instead.`

https://storybook.js.org/docs/react/api/csf#non-story-exports

### PR DESCRIPTION
**Pull request by yannbf**
This is a followup work on #12 as I couldn't push to the branch unfortunately. Original work by @i-udas

Closes #3, Closes #8, Closes #10, Closes #11

# What I did

- Updates Storybook example to version 7
- Updates implementation and dependency to support and use Storybook 7 apis

### 💥 Breaking Change

This version adds support for Storybook 7.0. It requires you to be using Storybook 7.0, as there were several internal changes required, all of which depend on new Storybook packages.

### 🚀 Features

In Storybook 7.0, the play function can also be defined in the Meta (default export). This is now supported in `@storybook/testing-vue3`.

Additionally, this package also supports CSF3, while maintaining compatibility with previous formats as well such as CSF1 and CSF2.

### ⚠️ Deprecations

The `setGlobalConfig` function is now deprecated in favor of `setProjectAnnotations`, which aligns better with Storybook 7.0 nomenclature.

**From:**
```js
  import { setGlobalConfig } from '@storybook/testing-vue3';
  import * as globalStorybookConfig from './.storybook/preview';

  setGlobalConfig(globalStorybookConfig);
```

**To:**
```js
  import { setProjectAnnotations } from '@storybook/testing-vue3';
  import * as globalStorybookConfig from './.storybook/preview';

  setProjectAnnotations(globalStorybookConfig);
```

#### The future of @storybook/testing-vue3

This package served as a way to provide testing utilities for Vue + Storybook users, and after some validation it's a candidate to be promoted into the `@storybook/vue3` package itself. Meaning, in the near future, the utilities of this package will be merged into `@storybook/vue3` and the `@storybook/testing-vue3` package will be deprecated, given that you won't need yet another package anymore. This should happen in the upcoming months.

Once this is done, there will be a final release notifying the users, and the migration should be simple and straightforward. Thank you so much for using this package!

<details>
  <summary>📦 Published PR as canary version: <code>1.0.0-canary.13.c3b6b8a.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/testing-vue3@1.0.0-canary.13.c3b6b8a.0
  # or 
  yarn add @storybook/testing-vue3@1.0.0-canary.13.c3b6b8a.0
  ```
</details>

### CODE REVIEW

**Comment by yannbf** [comment]
For anyone interested (cc @shuda323, @AlmarAubel, @mikiSumcul, @mrtnvh, @MichaelBitard, @Hanawa02, @i-udas), please follow this PR and test its canary version. 
It's currently in version `1.0.0-canary.13.922565e.0`, however with typescript issues for now (so ignore that part for now)

Feedback is deeply appreciated!

**Comment by yannbf** [comment]
@SocketSecurity ignore-all

**Comment by MichaelBitard** [comment]
Hi,

Thanks for your work!

We have tried this PR --> https://github.com/MTES-MCT/camino/pull/621/files#diff-1e6de3828ce1525bbc845baedb2831cd210cf0284863af1f7bbb618fc555c825

and we only have the `*.ts` storybook files failing, all our `*.tsx` storybook files work well

File: example/src/stories/Button.test.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import { Meta } from '@storybook/vue3';
+
+// example with composeStories, returns an object with all stories composed with args/decorators
+const { CSF3Primary } = composeStories(stories);
+
+// example with composeStory, returns a single story composed with args/decorators
+const Secondary = composeStory(stories.CSF2Secondary, stories.default);
+
+test('renders primary button', () => {
+  render(CSF3Primary({ label: 'Hello world' }));
```
Comment (chakAs3) [inline example/src/stories/Button.test.ts]: Hey @yannbf, I would like to clarify a few points here:

In Use-case 1, if we are testing against CFS Annotations using @testing-library/vue, everything looks good. 👍

In Use-case 2, if we want to test against the storybook renderer, we should use @storybook/testing-vue3 based on the implementation of @storybook/vue3 renderer.

At the moment, it seems we are in USE CASE 1: Passing the tests doesn't necessarily mean that @storybook/vue renders the story correctly. However, it validates the preview API and CSF annotation, and proves that delegating rendering to @testing-library/vue with the provided annotations works perfectly.

Actually, what we should do is test the @storybook/vue3 renderer again, so we need to expose the render function.

File: src/index.ts [hunk 2]
```diff
@@ -1,99 +1,134 @@
...
+  ComposedStory,
+  Store_CSFExports,
+  StoriesWithPartialProps,
+} from '@storybook/types';
+import { deprecate } from '@storybook/client-logger';
+import type { VueRenderer, Meta } from "@storybook/vue3";
 
-import decorateStory from "./decorateStory";
+//import { globalRender as render } from './utils';
+import { render } from './utils';  // use current implementation of render function in  @storybook/vue3
```
Comment (chakAs3) [inline src/index.ts]: to do a proper test we should expose the global render from  @storybook/vue3
```ts
import { render } from  '@storybook/vue3'
```
i have create a PR for that [#23529](https://github.com/storybookjs/storybook/pull/23529)

File: example/src/stories/Button.test.ts (hunk 1 above)
Comment (yannbf) [reply example/src/stories/Button.test.ts]: Hey @chakAs3 ! Thank you so much for checking this out. Although I agree that we should use the render function form @storybook/vue3, this package will likely be deprecated soon, in favor of moving this utility directly inside of the @storybook/vue3 package, just like it happened with @storybook/testing-react which moved to @storybook/react. Therefore I don't think we should expose the render function, if the only intention is to use it in this package

Comment (chakAs3) [reply example/src/stories/Button.test.ts]: OK cool, I agreed that this should move to monorepo, 😄 I was about to suggest this. for me, testing is a core feature of storybook, 
therefore I will just copy the code for now, and this fails 1 test due to the old issue I tried to explain many times,

 **_render function should return a representation of the DOM and not another function_** . 

this creates also miss behavior inside @storybook/vue3 renderer.  

I think we should ship this package for now even though is failing, ( cause this. is the purpose of it to test both CSF and @storybook/vue3 )

what do you suggest @yannbf?

Comment (yannbf) [reply example/src/stories/Button.test.ts]: Alright I figured it all out! Managed to get this working correctly by using all code from Storybook core libraries and the @storybook/vue3 annotations as well. I'll make a release and at some point start working on bringing this to the @storybook/vue3 package in the monorepo, so users can do `import { composeStories } from '@storybook/vue3'` and migrate away from this library.

Thank you so much @chakAs3, this helped me think this all through, especially the representation of the DOM part, which was crucial for the tests. <3

**Comment by yannbf** [comment]
> Hi,
> 
> Thanks for your work!
> 
> We have tried this PR --> https://github.com/MTES-MCT/camino/pull/621/files#diff-1e6de3828ce1525bbc845baedb2831cd210cf0284863af1f7bbb618fc555c825
> 
> and we only have the `*.ts` storybook files failing, all our `*.tsx` storybook files work well

Thank you so much @MichaelBitard, your reproduction was crucial to get this in the right shape. The most recent version should work with JSX and non-JSX stories: `1.0.0-canary.13.54193a7.0`, there's only one last step to get this 100%, which is the Typescript types!

**Comment by MichaelBitard** [comment]
You're welcome, thanks to you for doing all this! I was merely an alpha-tester :)

I updated our PR with your latest version and it works for tsx and ts files \o/.

I just had to add a function check `if (typeof value.story === 'function') {` here https://github.com/MTES-MCT/camino/pull/621/files#diff-1e6de3828ce1525bbc845baedb2831cd210cf0284863af1f7bbb618fc555c825R43 to make typescript happy

**Comment by yannbf** [comment]
> You're welcome, thanks to you for doing all this! I was merely an alpha-tester :)
> 
> I updated our PR with your latest version and it works for tsx and ts files \o/.
> 
> I just had to add a function check `if (typeof value.story === 'function') {` here https://github.com/MTES-MCT/camino/pull/621/files#diff-1e6de3828ce1525bbc845baedb2831cd210cf0284863af1f7bbb618fc555c825R43 to make typescript happy

Thanks once again <3
I believe I fixed all issues now, it's shippable and you wouldn't need any workaround. Could you please do a final test?

```
1.0.0-canary.13.c3b6b8a.0
```

**Comment by MichaelBitard** [comment]
hell yeah \o/
It's perfect!

**Comment by yannbf** [comment]
@MichaelBitard thanks once again, v1.0.0 is released!

**Pull request merged**